Error in `bazel run :rollup` when following the document

# 🐞 bug report

### Affected Rule

The issue is caused by the rule: `nodejs_binary`


### Is this a regression?

Not sure

### Description

Errors when trying out the example [usage](https://bazelbuild.github.io/rules_nodejs/#usage)

## 🔬 Minimal Reproduction

Follow the steps in https://bazelbuild.github.io/rules_nodejs/#usage until the step `bazel run:rollup -- --help`

## 🔥 Exception or Error
```sh
❯ bazel run :rollup --define=VERBOSE_LOGS=1 -- --help
INFO: Invocation ID: 3c1df155-1337-4567-9bc3-d6f074b16c8b
INFO: Analyzed target //:rollup (22 packages loaded, 71 targets configured).
INFO: Found 1 target...
INFO: Deleting stale sandbox base /private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/sandbox
Target //:rollup up-to-date:
  dist/bin/rollup.sh
  dist/bin/rollup_loader.js
  dist/bin/rollup_require_patch.js
INFO: Elapsed time: 1.940s, Critical Path: 0.03s
INFO: 0 processes.
INFO: Build completed successfully, 2 total actions
INFO: Build completed successfully, 2 total actions
[rollup_require_patch.js] patching require for //:rollup
  cwd: /private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup.sh.runfiles/bazel_nodejs_quickstart
  RUNFILES: /private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup.sh.runfiles
  TARGET: //:rollup
  BIN_DIR: bazel-out/darwin-fastbuild/bin
  GEN_DIR: bazel-out/darwin-fastbuild/bin
  INSTALL_SOURCE_MAP_SUPPORT: true
  MODULE_ROOTS: []
  NODE_MODULES_ROOT: build_bazel_rules_nodejs/node_modules
  USER_WORKSPACE_NAME: bazel_nodejs_quickstart

[rollup_require_patch.js] try to resolve in runfiles /private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup.sh.runfiles/bazel_nodejs_quickstart/node_modules/rollup/bin/rollup
[rollup_require_patch.js] try to resolve in runfiles /private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup.sh.runfiles/build_bazel_rules_nodejs/node_modules/bazel_nodejs_quickstart/node_modules/rollup/bin/rollup
Error: Cannot find module 'bazel_nodejs_quickstart/node_modules/rollup/bin/rollup'. Please verify that the package.json has a valid "main" entry
required in target //:rollup by 'undefined'
  looked in:
    built-in, relative, absolute, nested node_modules - Error: Cannot find module 'bazel_nodejs_quickstart/node_modules/rollup/bin/rollup'
Require stack:
-
    runfiles - Error: Cannot find module '/private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup.sh.runfiles/bazel_nodejs_quickstart/node_modules/rollup/bin/rollup'
Require stack:
-
    node_modules attribute (build_bazel_rules_nodejs/node_modules) - Error: Cannot find module '/private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup.sh.runfiles/build_bazel_rules_nodejs/node_modules/bazel_nodejs_quickstart/node_modules/rollup/bin/rollup'
Require stack:
-

    at Function.module.constructor._resolveFilename (/private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup_require_patch.js:482:17)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Object.<anonymous> (/private/var/tmp/_bazel_yujunz/a95322c336466e88acb40e8f0ae808b7/execroot/bazel_nodejs_quickstart/bazel-out/darwin-fastbuild/bin/rollup_loader.js:32:24)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11
```

## 🌍  Your Environment

**Operating System:**

macOS Catalina

**Output of `bazel version`:**

```
Build label: 2.1.1-homebrew
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Sat Feb 22 21:28:00 2020 (1582406880)
Build timestamp: 1582406880
Build timestamp as int: 1582406880
```

**Rules_nodejs version:**

(Please check that you have matching versions between WORKSPACE file and `@bazel/*` npm packages.)
```
    sha256 = "2eca5b934dee47b5ff304f502ae187c40ec4e33e12bcbce872a2eeb786e23269",
    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/1.4.1/rules_nodejs-1.4.1.tar.gz"],
```

**Anything else relevant?**

The repository for testing: https://github.com/yujunz/bazel_nodejs_quickstart/tree/rollup

--- a/docs/index.md
+++ b/docs/index.md
@@ -15,80 +15,134 @@ This JavaScript support lets you build and test code that targets a JavaScript r
 
 ## Quickstart
 
+First we create a new workspace, which will be in a new directory.
+We can use the `@bazel/create` npm package to do this in one command.
 This is the fastest way to get started for most use cases.
-See [the installation page](install.html) for details and alternative methods.
+
+> See [the installation page](install.html) for details and alternative methods.
 
 ```sh
-$ npm init @bazel
+$ npm init @bazel my_workspace
+$ cd my_workspace
 ```
 
-or if you prefer yarn,
+> You could do the same thing with yarn:
+> ```sh
+> $ yarn create @bazel my_workspace
+> $ cd my_workspace
+> ```
+> Both of these commands are equivalent to `npx @bazel/create` which downloads the latest version of the `@bazel/create` package from npm and runs the program it contains.
+> Run the tool with no arguments for command-line options and next steps.
+
+Next we install some development tools.
+For this example, we'll use Babel to transpile our JavaScript, Mocha for running tests, and http-server to serve our app.
+This is just an arbitrary choice, you probably already have some tools you prefer.
 
 ```sh
-$ yarn create @bazel
+$ npm install @babel/core @babel/cli @babel/preset-env http-server mocha domino
 ```
 
-> These commands are equivalent to `npx @bazel/create` which downloads the latest version of the `@bazel/create` package from npm and runs the program contained.
+Let's run these tools with Bazel. There are two ways to run tools:
 
-See the output of the tool for command-line options and next steps.
+- Use an auto-generated Bazel rule by importing from an `index.bzl` file in the npm package
+- Use a custom rule in rules_nodejs or write one yourself
 
-## Usage
-
-### Running a program from npm
-
-The `nodejs_binary` rule lets you run a program with Node.js.
-See [Built-ins](Built-ins.html)
-
-If you have installed the [rollup] package, you could write this rule:
+In this example we use the auto-generated rules.
+First we need to import them, using a load statement.
+So edit BUILD.bazel and add:
 
 ```python
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
-
-nodejs_binary(
-    name = "rollup",
-    entry_point = "//:node_modules/rollup/bin/rollup",
-)
+load("@npm//@babel/cli:index.bzl", "babel")
+load("@npm//mocha:index.bzl", "mocha_test")
+load("@npm//http-server:index.bzl", "http_server")
 ```
 
-and run it with
+> This shows us that rules_nodejs has told Bazel that a workspace named @npm is available
+> (think of the at-sign like a scoped package for Bazel).
+> rules_nodejs will add index.bzl files exposing all the binaries the package manager installed
+> (the same as the content of the node_modules/.bin folder).
+> The three tools we installed are in this @npm scope and each has an index file with a .bzl extension.
 
-```sh
-$ bazel run :rollup -- --help
-```
-
-[rollup]: https://www.npmjs.com/package/rollup
-
-You can also wrap an npm program with a Bazel rule, making it easy to integrate with a Bazel build.
-See the `examples/parcel` example.
-
-### Running a program from local sources
-
-We can reference a path in the local workspace to run a program we write.
+Next we teach Bazel how to transform our JavaScript inputs into transpiled outputs.
+Here we assume that you have `app.js` and `es5.babelrc` in your project. See [our example webapp](https://github.com/bazelbuild/rules_nodejs/tree/1.4.0/examples/webapp) for an example of what those files might look like.
+Now we want Babel to produce `app.es5.js` so we add to `BUILD.bazel`:
 
 ```python
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
-
-nodejs_binary(
-    name = "example",
+babel(
+    name = "compile",
     data = [
-        "@//:node_modules",
-        "main.js",
+        "app.js",
+        "es5.babelrc",
+        "@npm//@babel/preset-env",
     ],
-    entry_point = ":main.js",
-    args = ["--node_options=--expose-gc"],
+    outs = ["app.es5.js"],
+    args = [
+        "app.js",
+        "--config-file",
+        "$(execpath es5.babelrc)",
+        "--out-file",
+        "$(execpath app.es5.js)",
+    ],
 )
 ```
 
-This example illustrates how to pass arguments to nodejs (as opposed to passing arguments to the program).
+> This just calls the Babel CLI, so you can see [their documentation](https://babeljs.io/docs/en/babel-cli) for what arguments to pass.
+> We use the $(execpath) helper in Bazel so we don't need to hardcode paths to the inputs or outputs.
 
-The `data` attribute is optional, by default it includes the `node_modules` directory. To include your own
-sources, include a file or target that produces JavaScript.
+We can now build the application: `npm run build`
 
-See the `examples/user_managed_deps` directory in this repository.
+and we see the .js outputs from babel appear in the `dist/bin` folder.
 
-### Testing
+Let's serve the app to see how it looks, by adding to `BUILD.bazel`:
 
-The `examples/user_managed_deps/index.spec.js` file illustrates testing. Another usage is in https://github.com/angular/tsickle/blob/master/test/BUILD
+```
+http_server(
+    name = "server",
+    data = [
+        "index.html",
+        "app.es5.js",
+    ],
+    args = ["."],
+)
+```
+
+Add a `serve` entry to the scripts in `package.json`:
+
+```json
+{
+  "scripts": {
+    "serve": "ibazel run :server"
+  }
+}
+```
+
+> ibazel is the watch mode for bazel.
+>
+> Note that on Windows, you need to pass `--enable_runfiles` flag to Bazel.
+> That's because Bazel creates a directory where inputs and outputs both appear together, for convenience.
+
+Now we can serve the app: `npm run serve`
+
+Finally we'll add a test using Mocha, and the domino package so we don't need a browser. Add to `BUILD.bazel`:
+
+```python
+mocha_test(
+    name = "unit_tests",
+    args = ["*.spec.js"],
+    data = glob(["*.spec.js"]) + [
+        "@npm//domino",
+        "app.es5.js",
+    ],
+)
+```
+
+Run the tests: `npm test`
+
+## Next steps
+
+Look through the `/examples` directory in this repo for many examples of running tools under Bazel.
+
+You might want to look through the API docs for custom rules such as TypeScript, Rollup, and Terser which add support beyond what you get from calling the CLI of those tools.
 
 ### Debugging
 

